### PR TITLE
fix add puzzle callback

### PR DESCRIPTION
**Pull request by garber34**
Removed route.params from the dependency list in the callback in AddPuzzle to stop it from firing multiple times, which was causing puzzles to reload multiple times.

And added serviceaccount.ts to eslintignore because otherwise git wouldn't let me commit.

### CODE REVIEW

**Comment by adam-adair** [comment]
@garber34 Don't we want the callback to fire when the puzzle route changes because that means you're adding a new puzzle? 
It's been long enough that I don't remember why I added the route.params dependency, but I did so to fix a bug in [this commit](https://github.com/FJAM-Studios/pixtery/commit/670acc9edd7f0b236545ff2d3984c14d61c0f4ab). Probably should've documented that better. My guess: if the AddPuzzle screen is still mounted and you give it a new puzzle key, then it won't add it if there's no dependency.

Do we know why the route.params dependency is causing the callback to run multiple times? It seems like it shouldn't change, but I guess if the route _object_ is different for some reason, then it would run again. If I test it with the dependency of route.params.publickey instead of route.params, I don't get multiple firings of the callback, and it works as expected. That strikes me as safer than no dependency at all. Curious about what you think about why this happens in the first place, though; I might be misunderstanding something.

File: components/TransitionScreens/AddPuzzle.tsx [hunk 1]
```diff
@@ -172,7 +172,7 @@ export default function AddPuzzle({
         );
       };
       searchForPuzzle();
-    }, [route.params])
+    }, [])
```
Comment (adam-adair) [inline components/TransitionScreens/AddPuzzle.tsx:175]: See my comment about making this route.params.publicKey instead.

**Comment by garber34** [comment]
@adam-adair honestly I don't know why route params would be changing like that. 

But I am still able to add multiple puzzles when removing it from the dependency list

**Comment by adam-adair** [comment]
@garber34 The nav.replace idea seemed plausible to me, but then when I comment it out, still get the same issue.
I am able to add multiple puzzles with no dependencies and with route.params.publicKey dependency, too, so I don't really know what my supposed bug fix was about. I am fine to merge this and file it under 'keep an eye on it'.

**Pull request merged**